packages/react-liveness/src/components/FaceLivenessDetector/service/machine/machine.ts

```diff
@@ -995,63 +995,44 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
       },
       async detectFaceDistance(context) {
         const {
+          serverSessionInformation,
           isFaceFarEnoughBeforeRecording: faceDistanceCheckBeforeRecording,
         } = context;
-        const { videoEl, videoMediaStream, isMobile } =
-          context.videoAssociatedParams!;
+        const { videoEl, videoMediaStream } = context.videoAssociatedParams!;
         const { faceDetector } = context.ovalAssociatedParams!;
 
         const { width, height } = videoMediaStream!
           .getTracks()[0]
           .getSettings();
 
+        const challengeConfig =
+          serverSessionInformation?.Challenge?.FaceMovementAndLightChallenge
+            ?.ChallengeConfig;
+
+        if (!challengeConfig || !challengeConfig.OvalHeightWidthRatio) {
+          throw new Error(
+            'Challenge config not returned from session information.'
+          );
+        }
```

Comment:
How does the machine reach this state without the `challengeConfig` being defined and what is the impact on the end user UX?

Reply:
I believe for the most part this case should never be reached. We have other checks before this to ensure that the session information exists. This is here in the case the server decides to return a session information without challenge config, which would be an error on the rekognition service side. In the case that this is triggered they end user would see the usual error experience and the developer could print this error from the `onError` cb

Reply:
Thanks for the explanation! Noted [here](https://github.com/aws-amplify/amplify-ui/pull/5254/files#diff-4dc9d74828ef070874870649a51d5a4dd622a7ab2eca61a4d761f36075cffd9dR657) as well, but these validations could be run against `challengeConfig` before setting in the machine?